src/c++/perf_analyzer/test_concurrency_manager.cc

```diff
@@ -667,4 +678,113 @@ TEST_CASE("Concurrency - Shared memory methods")
   }
 }
 
+TEST_CASE("concurrency_deadlock")
+{
+  PerfAnalyzerParameters params{};
+  params.max_concurrency = 6;
+  bool is_sequence_model{true};
+  bool some_infer_failures{false};
+
+  const auto& ParameterizeSync{[&]() {
+    SUBCASE("sync")
+    {
+      params.async = false;
+      params.streaming = false;
+    }
+    SUBCASE("aync no streaming")
+    {
+      params.async = true;
+      params.streaming = false;
+    }
+    SUBCASE("async streaming")
+    {
+      params.async = true;
+      params.streaming = true;
+    }
+  }};
+
+  const auto& ParameterizeConcurrency{[&]() {
+    SUBCASE("1 concurrency, 1 thread")
+    {
+      ParameterizeSync();
+      params.max_concurrency = 1;
+      params.max_threads = 1;
+    }
+    SUBCASE("4 concurrency, 4 thread")
+    {
+      ParameterizeSync();
+      params.max_concurrency = 4;
+      params.max_threads = 4;
+    }
+    SUBCASE("4 concurrency, 3 thread")
+    {
+      ParameterizeSync();
+      params.max_concurrency = 4;
+      params.max_threads = 3;
+    }
+    SUBCASE("4 concurrency, 1 thread")
+    {
+      ParameterizeSync();
+      params.max_concurrency = 4;
+      params.max_threads = 1;
+    }
+  }};
+
+  const auto& ParameterizeSequence{[&]() {
+    SUBCASE("non-sequence")
+    {
+      ParameterizeConcurrency();
+      is_sequence_model = false;
+    }
+    SUBCASE("sequence")
+    {
+      ParameterizeConcurrency();
+      is_sequence_model = true;
+      params.num_of_sequences = 3;
```

Comment:
I don't think num_of_sequences is used for concurrency mode. I believe PA creates N number of sequences based on the concurrency value. In other words, I think it's an invalid configuration to specify num_of_sequences that is different from max_concurrency

Reply:
I removed it.